src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ViewDeclarationOrder/ViewDeclarationOrderAnalyzer.cs

```diff
@@ -1,77 +1,124 @@
-		}
+		public ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Descriptors.PX1004_ViewDeclarationOrder, Descriptors.PX1006_ViewDeclarationOrder);
 
-		private static void AnalyzeGraphViews(SymbolAnalysisContext context, PXContext pxContext)
+		public void Analyze(SymbolAnalysisContext symbolContext, PXContext pxContext, CodeAnalysisSettings settings,
+							PXGraphSemanticModel graphSemanticModel)
 		{
-			INamedTypeSymbol graph = (INamedTypeSymbol)context.Symbol;
-
-			if (graph == null || context.CancellationToken.IsCancellationRequested ||
-			   !graph.InheritsFromOrEquals(pxContext.PXGraph.Type) && !graph.InheritsFromOrEquals(pxContext.PXGraphExtensionType))
-			{
-				return;
-			}
-        
-            var graphViews = graph.GetMembers()
-								  .OfType<IFieldSymbol>()
-								  .Select(field => field.Type as INamedTypeSymbol)
-								  .Where(fieldType => fieldType != null &&
-													   fieldType.InheritsFrom(pxContext.PXSelectBase.Type) && 
-													   fieldType.IsGenericType &&
-													   fieldType.TypeArguments.Length > 0).
-								   ToImmutableList();
-
-			Location graphLocation = graph.Locations.FirstOrDefault();
-
-			if (graphLocation == null || context.CancellationToken.IsCancellationRequested)
+			if (graphSemanticModel.ViewsByNames.Count == 0 || IsNewMethodUsedToInitCaches(pxContext))
 				return;
 
-			//forward pass
-			CheckGraphViews(context, graphLocation, graphViews, Descriptors.PX1004_ViewDeclarationOrder);
+			symbolContext.CancellationToken.ThrowIfCancellationRequested();
 
-			if (context.CancellationToken.IsCancellationRequested)
+			var viewsGroupedByDAC = GetViewsUsedInAnalysis(graphSemanticModel).Where(view => view.ViewDAC != null)
+																			  .ToLookup(view => view.ViewDAC);
+			if (viewsGroupedByDAC.Count == 0)
 				return;
 
-			//backward pass
-			CheckGraphViews(context, graphLocation, graphViews.Reverse(), Descriptors.PX1006_ViewDeclarationOrder);
+			foreach (IGrouping<ITypeSymbol, DataViewInfo> dacViews in viewsGroupedByDAC)
+			{
+				AnalyzeDacViewsForNumberOfCaches(graphSemanticModel, symbolContext, dacViews, viewsGroupedByDAC);
+			}
+
+			symbolContext.CancellationToken.ThrowIfCancellationRequested();
 		}
 
-		private static void CheckGraphViews(SymbolAnalysisContext context, Location graphLocation, ImmutableList<INamedTypeSymbol> graphViews,
-											DiagnosticDescriptor diagnosticDescriptor)
+		/// <summary>
+		/// Starting from the Acumatica 2018R2 version a new method is used to initialize caches with explicit ordering of caches.
+		/// </summary>
+		/// <returns/>
+		private static bool IsNewMethodUsedToInitCaches(PXContext pxContext)
 		{
-			var visitedViews = new HashSet<ITypeSymbol>();
+			IMethodSymbol initCachesNewMethod = pxContext.PXGraph.Type
+																 .GetMembers(InitCacheMappingMethodName)
```

Comment:
In such cases we place symbols to the appropriate utility classes. In this case, you can use `Acuminator.Utilities.Roslyn.Semantic.Symbols.PXGraphSymbols` class to store the method symbol.

Reply:
I wasn't sure - because this symbol  probably will be used by only this diagnostic. But I agree - it's cleaner this way. Will fix